service/service_account.go

```diff
@@ -64,14 +64,55 @@ func (s AccountService) AccountBalance(
 		return nil, wrapError(errInternalError, metadataErr)
 	}
 
+	avaxBalance, balanceErr := s.client.BalanceAt(context.Background(), address, header.Number)
+	if balanceErr != nil {
+		return nil, wrapError(errInternalError, balanceErr)
+	}
+
+	var balances []*types.Amount
+	balances = append(balances, mapper.AvaxAmount(avaxBalance))
```

Comment:
You should only add `avaxBalance` here if `len(req.Currencies)` == 0, otherwise, you should only populate it if it is in `req.Currencies`.